src/redisearch_rs/wildcard/src/lib.rs

```diff
@@ -245,18 +247,21 @@ impl<'pattern> WildcardPattern<'pattern> {
                     bt_state = Some((i_t, i_k));
                 }
                 Token::Literal(chunk) => {
-                    for i in 0..chunk.len() {
-                        let Some(key_char) = key.get(i_k + i) else {
-                            // It may have matched if we had more characters in the key
-                            return MatchOutcome::PartialMatch;
-                        };
-                        if &chunk[i] != key_char {
-                            try_backtrack!(bt_state, i_t, i_k, 'outer);
-                        }
+                    let remaining_key_len = key.len() - i_k;
+                    let (min_len, is_partial_match) = if chunk.len() > remaining_key_len {
+                        (remaining_key_len, true)
+                    } else {
+                        (chunk.len(), false)
+                    };
+                    if !is_prefix(&key[i_k..], &chunk[..min_len]) {
+                        try_backtrack!(bt_state, i_t, i_k, 'outer);
+                    }
+                    if is_partial_match {
+                        return MatchOutcome::PartialMatch;
+                    } else {
+                        i_t += 1;
+                        i_k += chunk.len();
                     }
```

Comment:
nitpick: with early exit in the if block, `else` block becomes redundant and best not to nest when not needed for readability:

```suggestion
                    if is_partial_match {
                        return MatchOutcome::PartialMatch;
                    }

                    i_t += 1;
                    i_k += chunk.len();
```